difflint/lib.py

```diff
@@ -205,6 +229,73 @@ def detect_new_diff_lint_errors(log_output):
     return False
 
 
+def save_merge_state():
+    """
+    Saves the merging state in the event that we're resolving
+    a merge conflict. Stashing will mangle the merge state unless
+    we have saved it ahead of time. The reverse operation to this
+    function is load_merge_state().
+
+    Special thanks to the overcommit project for solving this
+    problem in a similar context. (https://github.com/brigade/overcommit)
+
+    Inputs: None
+
+    Output: If we were are in a merge conflict state, this will return a
+            tuple containing first the merge message, and second the
+            merge commit hash.
+            If we were not in a merge conflict state, this will return a
+            tuple containing first an empty string, and second another
+            empty string.
+    """
+    merge_head_commit_hash = ''
+    try:
+        with open(devnull, 'w') as DEV_NULL:
+            # This will fail if MERGE_HEAD doesn't exist.
+            merge_head_commit_hash = check_output(['git', 'rev-parse', 'MERGE_HEAD'],
+                                                  stderr=DEV_NULL).decode()
+    except CalledProcessError:
+        return ('', '')
+
+    merge_msg = ''
+    merge_msg_filename = make_path_from_cwd('.git', 'MERGE_MSG')
+    if os.path.isfile(merge_msg_filename):
+        with open(merge_msg_filename, 'r') as f:
+            merge_msg = f.read()
+    
+    return (merge_msg, merge_head_commit_hash)
+
+
+def load_merge_state(merge_msg, merge_head_commit_hash):
+    """
+    Restores the merge state saved by a previous call to
+    save_merge_state(). This function should only be called
+    if save_merge_state() returned a non-empty string.
+
+    Inputs: 
+        merge_msg: The saved message from a merge conflict
+            
+        merge_head_commit_hash: The hash belonging to the
+            MERGE_HEAD reference.
+
+    Output: None
+    """
+    merge_mode_filename = make_path_from_cwd('.git', 'MERGE_MODE')
```

Comment:
Actually, all of the calls to `make_path_from_cwd` are unnecessary like the one above, so `make_path_from_cwd` itself can be removed.